### PR TITLE
Refactor Readme to incorporate latest feature set

### PR DESCRIPTION
**Pull request by gkielian**
This README contains a quickstart for installation, data prep, training, smapling, and the flow for running explorations, on GPU.

Currently we have maintained compatibility as much as possible, but just to be safe, removing the sections we have not yet tested after refactoring the `train.py` and `sample.py` scripts to utilize argparse for parameterization.

There is a `TODO` added for a colab, which should be filled in once the full url is available (maybe with the open in colabs button).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,50 +1,103 @@
-
 # nanoGPT
```
Comment (msaligane) [inline README.md:1]: Should this be renamed?

Comment (gkielian) [reply README.md:1]: Good point, I'll update.

Comment (gkielian) [reply README.md]: @msaligane All set

**Comment by gkielian** [comment]
Looks like I'll have to rebase this, please hold before merging.

**Comment by gkielian** [comment]
Commit chain is fixed now, should be only one commit now and the two markdown files

**Review by msaligane — APPROVED** [review]

**Pull request merged**